Services/BetQuoteService.cs

```diff
@@ -1,34 +1,74 @@
 ﻿using Services.Interfaces;
 using DataAccess;
-using Microsoft.EntityFrameworkCore;
 using Domain.Entities;
+using Domain.Dto.BetQuote;
+using AutoMapper;
+using System.Collections.Generic;
 
 namespace Services
 {
     public sealed class BetQuoteService : IBetQuoteService
     {
         private readonly DBContext _dbContext;
+        private readonly IMapper _mapper;
+        private readonly IBetsService _betsService;
 
-        public BetQuoteService(DBContext dBContext)
+        public BetQuoteService(DBContext dBContext, IMapper mapper, IBetsService betsService)
         {
             _dbContext = dBContext;
+            _mapper = mapper;
+            _betsService = betsService;
         }
 
-        public async Task<BetQuotes> CreateAsync(BetQuotes entity)
-        {   
-            _dbContext.BetQuotes.Add(entity);
+        public async Task<BetQuoteDto> CreateAsync(CreateBetQuotesDto entity, Guid betId)
+        {
+            BetQuotes newBetQuote = _mapper.Map<BetQuotes>(entity);
+            newBetQuote.BetId = betId;
+            _dbContext.BetQuotes.Add(newBetQuote);
             await _dbContext.SaveChangesAsync();
-            return entity;
+            BetQuoteDto newBetQuoteDto = _mapper.Map<BetQuoteDto>(newBetQuote);
+            return newBetQuoteDto;
+        }
+
+        public IQueryable<BetQuoteDto> GetAllAsync()
+        {
+            IEnumerable<BetQuoteDto> entities = _mapper.Map<IEnumerable<BetQuoteDto>>(_dbContext.BetQuotes);
```

Comment:
the mapper will materialize the query
you need to return the entity, even if it will be exposed in the facade